### PR TITLE
[Tests-Only] Fix timing problem in testScanShareInvalidLastScan

### PR DESCRIPTION
**Pull request by phil-davis**
## Description
https://drone.owncloud.com/owncloud/core/26677/35/9
```
There was 1 failure:

1) OCA\Files_Sharing\Tests\External\ScanExternalSharesJobTest::testScanShareInvalidLastScan
Failed asserting that true matches expected false.

/drone/src/apps/files_sharing/tests/External/ScanExternalSharesJobTest.php:281
```

If the system time ticks over the 1-second boundary just after the test has remembered the time in to return in the mock of `getLastLogin` and the time that is put in `lastscan`, then `$now` in `shouldScan` becomes 1 second later. This happens to make both reasons to return false be false, and `shouldScan` unexpectedly returns true.

Putting a \sleep(1); just before:
`$this->invokePrivate($scanShares, 'shouldScan', [$share, $lastLoginThreshold, $lastScanThreshold]);`
Makes the test fail consistently.

Adjust `$lastScanThreshold` so that an accidental 1-second change in the system time will not make the test fail.

## How Has This Been Tested?
Local unit test runs

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:
- [ ] Code changes
- [ ] Unit tests added
- [ ] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: apps/files_sharing/tests/External/ScanExternalSharesJobTest.php [hunk 1]
```diff
@@ -275,7 +275,7 @@ public function testScanShareInvalidLastScan() {
 			'owner'	=> 'test'
 		];
 		$lastLoginThreshold = '1';
-		$lastScanThreshold = '1';
+		$lastScanThreshold = '2';
```
Comment (phil-davis) [inline apps/files_sharing/tests/External/ScanExternalSharesJobTest.php:278]: Without this change, adding `\sleep(1)` here makes the test fail.
An unlucky tick of the system clock between setting `'lastscan' => \time(),` at line 274, and getting into the the code of `shouldScan` causes the test to fail.

Comment (mrow4a) [reply apps/files_sharing/tests/External/ScanExternalSharesJobTest.php:278]: perfectly fine, most important is that it is larger :) could be even larger, as `\time()+2` could still see issue.

Plus would be good to add -> `add 10 seconds to time() in order to force shouldScan return false`

**Review by mrow4a — CHANGES_REQUESTED** [review]

File: apps/files_sharing/tests/External/ScanExternalSharesJobTest.php (hunk 1 above)
Comment (phil-davis) [reply apps/files_sharing/tests/External/ScanExternalSharesJobTest.php:278]: Or I could make a more realistic "ordinary" case - set `$lastScanThreshold = '20'`  and `'lastscan' => \time() - 10,`

A share that has been scanned 10 seconds ago, and the last scan threshold is 20 seconds, should not be chosen for scanning.

The existing test here was checking exactly the edge case of the last scan time being just 1 second different from the threshold. But actually for this feature we don't care about that sort of edge case. e.g. if the threshold is set to 900 seconds (15 minutes) then if a share was last scanned 899, 900, or 901 seconds ago we really do not care if the one at exactly 900 seconds is chosen for scanning or not. The requirement does not need to specify and test that.

Comment (phil-davis) [reply apps/files_sharing/tests/External/ScanExternalSharesJobTest.php]: I did what I wrote above ^

**Comment by mrow4a** [comment]
@phil-davis shalll we do the same with login ?

**Comment by phil-davis** [comment]
> @phil-davis shall we do the same with login ?

`testScanShareInvalidLastLogin` does not have the same timing issue. It sets the user to have last logged in 2 seconds ago, then sets the last login threshold to 1 second, then calls `shouldScan`. If there is 1 or more seconds delay before `shouldScan` runs then it makes no difference to the test result. The user will look like they last logged in 3, 4, 5... seconds ago, the threshold is 1 second, so they will not be considered a "recent" user.

Do you think I should adjust those numbers anyway to be more "realistic"?

**Comment by phil-davis** [comment]
@mrow4a or anyone, please review

**Review by mrow4a — APPROVED** [review]
